scripts/linters/pylint_extensions.py

```diff
@@ -2109,6 +2109,46 @@ def visit_call(self, node):
                     break
 
 
+class DisallowBlankLinesBelowFunctionDefinitionChecker(checkers.BaseChecker):
+    """Custom pylint checker to ensure there are no blank lines
+    below function definition.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'disallow-blank-lines-below-function-definition'
+    priority = -1
+    msgs = {
+        'C0034': (
+            'Please remove the blank line(s) below the function definition.',
+            'remove-blank-lines-below-function-definiton.',
+            'Disallow blank lines below function definition.'
+        ),
+    }
+
+    def visit_functiondef(self, node):
+        """Visits every function definition and ensures there are no blank lines
+        below function definition.
+
+        Args:
+            node: astroid.scoped_nodes.Function. Node for a function or method
+                definition in the AST.
+        """
+        c = 0
+        for i in python_utils.RANGE(len(node.root().file)):
```

Comment:
The `node.root().file` is the filname or the file content?

Reply:
This gives filename, Very sorry I did this also by mistake. I played around with various functions like [tolineno](https://www.kite.com/python/docs/astroid.node_classes.NodeNG.tolineno),  [block_range](http://pylint.pycqa.org/projects/astroid/en/latest/_modules/astroid/scoped_nodes.html#FunctionDef.block_range) and a few others, but none of them helped me in getting the last line number of node. Basically, the logic is to iterate through the lines of node untill the first colon, first colon appears at the end of function definition.  This was easier with while loop as it does not need any predefined range, but with for loop i have to enter a range.  So I have used 999 as the ending range, I have assumed that no function defintion is greater than 999 lines and I think this is a fair assumption. Moreover if the colon is found earlier like in one or two lines, for loop will break. So its not like loop will iterate 999 times for every function,

Reply:
@DubeySandeep Please guide me if I should proceed with this or should try something else?